bundle/config/mutator/initialize_workspace_client.go

```diff
@@ -20,7 +21,22 @@ func (m *initializeWorkspaceClient) Name() string {
 // Apply initializes the workspace client for the bundle. We do this here so
 // downstream calls to b.WorkspaceClient() do not panic if there's an error in the
 // auth configuration.
-func (m *initializeWorkspaceClient) Apply(_ context.Context, b *bundle.Bundle) diag.Diagnostics {
+func (m *initializeWorkspaceClient) Apply(ctx context.Context, b *bundle.Bundle) diag.Diagnostics {
+	if err := validateWorkspaceHost(ctx, b); err != nil {
+		return err
+	}
+
 	_, err := b.InitializeWorkspaceClient()
 	return diag.FromErr(err)
 }
+
+func validateWorkspaceHost(ctx context.Context, b *bundle.Bundle) diag.Diagnostics {
+	env_host, _ := env.Host(ctx)
+	target_host := b.Config.Workspace.Host
+
+	if env_host != "" && target_host != "" && env_host != target_host {
+		return diag.Errorf("target host and DATABRICKS_HOST environment variable mismatch")
```

Comment:
Could you explain why would it matter if there's a mismatch?

Host can be overriden in bundle config for example and that's where it's going to be taken from even if DATABRICKS_HOST is specified, so DATABRICKS_HOST is going to be ignored, so why would it matter?

With this change, CLI won't authenticate anymore, unless I am missing something

Reply:
It matters in the workspace and the web terminal because they deploy to the current workspace (DATABRICKS_HOST), so the bundle config has to match.

Maybe @ilia-db can add some context here.

Reply:
> DATABRICKS_HOST is going to be ignored, so why would it matter?

The problem is with other auth env vars that are set together with the host, in the webterminal and workspace case it's DATABRICKS_TOKEN. Since the token is set for the specific host, using anything else results in a 'bad token' errors.

Maybe we can be more explicit here and only error out for a host mismatch when DATABRICKS_TOKEN exists?

Reply:
We should not error here because it'll be a breaking change for users.  We can show a warning instead that two different host URLs have been provided, and the one from the bundle configuration is being used.

Reply:
After discussion we've agreed that improving the the json-parsing error that we get when the token is wrong might be a better approach